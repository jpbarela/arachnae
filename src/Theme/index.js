// @flow
export type ButtonThemeType = {
  primary: string,
};

export type ThemeType = {
  background: string,
  button: ButtonThemeType,
  color: string,
  fontFamily: string,
};

export const defaultTheme: ThemeType = {
  button: {
    primary: "blue",
  },
  color: "black",
  background: "white",
  fontFamily: "Helvetica, sans-serif",
};
