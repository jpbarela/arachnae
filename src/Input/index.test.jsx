// @flow
import * as React from "react";
import { render } from "@testing-library/react";
import { DateInput, Input, NumericInput } from "./index";
import { ThemeProvider } from "react-jss";
import { defaultTheme } from "../Theme";

describe("Inputs", () => {
  describe("Input", () => {
    it("renders", () => {
      const container = render(
        <ThemeProvider theme={defaultTheme}>
          <Input />
        </ThemeProvider>
      );
      expect(container.baseElement.firstChild).toMatchSnapshot();
    });
  });

  describe("NumericInput", () => {
    it("renders", () => {
      const container = render(
        <ThemeProvider theme={defaultTheme}>
          <NumericInput />
        </ThemeProvider>
      );
      expect(container.baseElement.firstChild).toMatchSnapshot();
    });
  });

  describe("DateInput", () => {
    it("renders", () => {
      const container = render(
        <ThemeProvider theme={defaultTheme}>
          <DateInput />
        </ThemeProvider>
      );
      expect(container.baseElement.firstChild).toMatchSnapshot();
    });
  });
});
