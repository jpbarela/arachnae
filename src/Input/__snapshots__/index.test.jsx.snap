// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs Input renders 1`] = `
<div>
  <input
    class="input-0-2-1"
    value=""
  />
</div>
`;

exports[`Inputs NumericInput renders 1`] = `
<div>
  <input
    class="input-0-2-1"
    type="number"
    value=""
  />
</div>
`;
